src/Module.ts

```diff
@@ -471,24 +470,30 @@ export default class Module {
 
 	private getOriginalLocation(
 		sourcemapChain: RawSourceMap[],
-		location: { line: number; column: number }
+		location: { line: number; column: number; source?: string; name?: string }
 	) {
-		const filteredSourcemapChain = sourcemapChain
-			.filter(sourcemap => sourcemap.mappings)
-			.map(sourcemap => {
-				const encodedSourcemap = sourcemap;
-				if (sourcemap.mappings) {
-					encodedSourcemap.mappings = encode(encodedSourcemap.mappings);
-				}
-				return encodedSourcemap;
-			});
+		const filteredSourcemapChain = sourcemapChain.filter(sourcemap => sourcemap.mappings);
+
 		while (filteredSourcemapChain.length > 0) {
 			const sourcemap = filteredSourcemapChain.pop();
-			const smc = new SourceMapConsumer(sourcemap);
-			location = smc.originalPositionFor({
-				line: location.line,
-				column: location.column
-			});
+			const line = sourcemap.mappings[location.line - 1];
+			let locationFound = false;
+
+			for (const segment of line) {
+				if (+segment[0] >= location.column) {
```

Comment:
Is the `+` coercion really necessary?

Reply:
Yes, it complains about values being strings. The original type issue here is that source map objects are typed as `RawSourceMap`, but `mappings` is actually decoded into arrays (while TypeScript assumes it's a single string). I'd suggest fixing the typing here in a separate type-cleaning PR because a proper fix might be quite involved.